### PR TITLE
refactor(input): Migrate Input component to Ant Design 5

### LINKED ISSUE #30727: Ant Design 5: Upgrade Input component
Please refer to SIP-139 https://github.com/apache/superset/issues/29268.

The `Input` component should be upgraded to Ant Design version 5.

Similar work has been completed [here](https://github.com/apache/superset/pulls?q=is%3Apr+label%3Afrontend%3Arefactor%3Aantd5+is%3Aclosed).

When upgrading the component to Ant Design 5, please ensure to include:

- Proper RTL test coverage if not present yet.
- A Storybook file (if not present) or all necessary changes.
- Include all relevant component options in the Storybook.
- Clean up unnecessary custom styles.

When opening a PR:
- Name your PR refactor({Component}): Upgrade {component} to Ant Design 5
- Add screenshots of BEFORE/AFTER in your PR.
- Update the [project tracker](https://github.com/apache/superset/discussions/29293) when your PR is merged.
- Make sure you or a committer has labeled your PR with [frontend:refactor:antd5](https://github.com/apache/superset/pulls?q=is%3Apr+label%3Afrontend%3Arefactor%3Aantd5+is%3Aclosed).

### PR DESCRIPTION
**Pull request by msyavuz**
### SUMMARY

Migrate Input Component to Ant Design 5 while keeping the visuals consistent.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
Standalone:
input-b-s

Inside SqlLab Save query:
input-b-i

After:
Standalone:
input-a-s

Inside SqlLab Save query:
input-a-i

### TESTING INSTRUCTIONS

Test component visuals before and after in used places or in storybook.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #30727 
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx [hunk 1]
```diff
@@ -138,10 +138,9 @@ export const DrillByMenuItems = ({
 
   useEffect(() => {
     if (open) {
-      ref.current?.input.focus({ preventScroll: true });
+      ref.current?.input?.focus({ preventScroll: true });
     } else {
       // Reset search input when menu is closed
-      ref.current?.setValue('');
```
Comment (geido) [inline superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:144]: We don't need this anymore? It seems unrelated to the upgrade

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx [hunk 2]
```diff
@@ -311,6 +310,7 @@ export const DrillByMenuItems = ({
                 margin: ${theme.gridUnit * 2}px ${theme.gridUnit * 3}px;
                 box-shadow: none;
               `}
+              value={searchInput}
```
Comment (geido) [inline superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:313]: Same for this. It seems unrelated to the upgrade? What is this for?

File: superset-frontend/src/explore/components/controls/BoundsControl.test.jsx [hunk 3]
```diff
@@ -34,7 +34,8 @@ test('renders two inputs', () => {
 test('receives null on non-numeric', async () => {
   render(<BoundsControl {...defaultProps} />);
   const minInput = screen.getAllByRole('spinbutton')[0];
-  userEvent.type(minInput, 'text');
+  userEvent.type(minInput, '1');
+  userEvent.clear(minInput);
```
Comment (geido) [inline superset-frontend/src/explore/components/controls/BoundsControl.test.jsx:38]: I think this will be giving a false positive for this test?

File: superset-frontend/src/features/alerts/AlertReportModal.tsx [hunk 4]
```diff
@@ -983,8 +983,15 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
...
 
-  const onCustomWidthChange = (value: number | null | undefined) => {
-    updateAlertState('custom_width', value);
+  const onCustomWidthChange = (value: number | string | null | undefined) => {
+    // Convert string to number if needed and handle empty/invalid values
+    const numValue =
+      value === undefined || value === ''
+        ? undefined
+        : typeof value === 'string'
+          ? Number(value) || null
```
Comment (geido) [inline superset-frontend/src/features/alerts/AlertReportModal.tsx:992]: Why are we defaulting to `null` here? What happens if value is '0'?

File: superset-frontend/src/features/alerts/components/AlertReportCronScheduler.tsx [hunk 5]
```diff
@@ -51,17 +51,15 @@ export const AlertReportCronScheduler: FC<AlertReportCronSchedulerProps> = ({
   onChange,
 }) => {
   const theme = useTheme();
-  const inputRef = useRef<AntdInput>(null);
```
Comment (geido) [inline superset-frontend/src/features/alerts/components/AlertReportCronScheduler.tsx:54]: Is this not necessary after the upgrade?

**Review by geido — COMMENTED** [review]
This is great! A few clarifications needed

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx (hunk 1 above)
Comment (msyavuz) [reply superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:144]: There is no setValue method for Input after the migration so this would give type errors.

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx (hunk 2 above)
Comment (msyavuz) [reply superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:313]: This is said binding of the state.

File: superset-frontend/src/explore/components/controls/BoundsControl.test.jsx (hunk 3 above)
Comment (msyavuz) [reply superset-frontend/src/explore/components/controls/BoundsControl.test.jsx:38]: InputNumber only accepts numbers by default on v5 so typing 'text' does not fire onChange. This test might be obsolete alltogether.

File: superset-frontend/src/features/alerts/AlertReportModal.tsx (hunk 4 above)
Comment (msyavuz) [reply superset-frontend/src/features/alerts/AlertReportModal.tsx]: That's a good catch.

File: superset-frontend/src/features/alerts/components/AlertReportCronScheduler.tsx (hunk 5 above)
Comment (msyavuz) [reply superset-frontend/src/features/alerts/components/AlertReportCronScheduler.tsx:54]: setValue is obsolete and redundant on the customSetValue callback and i think handlePressEnter can use value directly so this becomes unneccessary.

File: superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx (hunk 2 above)
Comment (geido) [reply superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx:313]: I think this approach does not work. Watch the example video. I cannot even type right, it will lag really bad. 


https://github.com/user-attachments/assets/c94d51e2-36fb-4b92-8677-303a83accdac

**Review by geido — APPROVED** [review]
LGTM!
